transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpAcceptProcessor.java

```diff
@@ -361,7 +361,7 @@ public static void setServerHeader(DefaultHttpSession httpSession, HttpResponseM
         if (httpSession != null) {
             ResourceAddress address = httpSession.getLocalAddress();
             boolean serverHeaderEnabled = address.getOption(HttpResourceAddress.SERVER_HEADER_ENABLED);
-            if (serverHeaderEnabled) {
+            if (serverHeaderEnabled && !httpResponse.hasHeader("Server")) {
```

Comment:
Good point. I was thinking about it too. I thought there would be ways to configure at the origin server level. Moreover, I think more generic solution to configure any header.